flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -120,6 +132,34 @@ public void closeOpenChannelConnections(ConnectionID connectionId) {
 		}
 	}
 
+	void connectChannelWithRetry(ConnectingChannel connectingChannel, ConnectionID connectionId)
+		throws IOException, InterruptedException {
+		int count = 0;
+		Exception exception = null;
+		do {
+			try {
+				connectingChannel = new ConnectingChannel(connectionId, this);
+				LOG.info("Connecting to {} at {} attempt", connectionId.getAddress(), count);
+				nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
+
+				NettyPartitionRequestClient client = connectingChannel.waitForChannel();
+
+				clients.replace(connectionId, connectingChannel, client);
+			} catch (IOException e) {
+				LOG.error("Failed {} times to connect to {}", count, connectionId.getAddress(), e);
+				exception = e;
+			} catch (ChannelException e) {
```

Comment:
These two exceptions can be merged together because the handling logic is the same.